### ISSUE TITLE
Upgrade from 3.0.0-beta7 to 3.0.0 cause babel-generator fails

### ISSUE THREAD

**Issue by Meeeeow** [gaearon/react-hot-loader#689]
### Description
Upgrade from 3.0.0-beta7 to 3.0.0 cause babel-generator fails:

```
Cannot read property '0' of null
    at Buffer._append (C:\Users\***\node_modules\babel-generator\lib\buffer.js:95:25)
```

### Expected behavior
No errors

### Actual behavior
Cause babel-generator crash with special code

### Environment

React Hot Loader version: 3.0.0

Run these commands in the project folder and fill in their results:

1. `node -v`: 8.5.0
2. `npm -v`: 5.3.0

Then, specify:

1. Operating system: Windows 10
2. Browser and version: Compile failed

### Reproducible Demo
Use `gaearon/react-hot-boilerplate/tree/next` and change react-hot-loader to `^3.0.0`, then add code listed below to App.js
```js
class B {
  constructor () {
    this['dynamic'] = () => {

    }
  }
}
```

**Comment by theKashey** [comment]
This is related to #686. We are on track.

**Comment by neoziro** [comment]
@theKashey can you contact me on Twitter or by email, so we could talk directly on Slack or Hangout or ...

**Comment by neoziro** [comment]
Closed as duplicate of #686
